scripts/meta_analysis.py

```diff
@@ -553,9 +566,26 @@ def get_next_variant( studies : List[Study]) -> List[VariantData]:
             res.append(None)
             continue
 
+        # Flag tracks that only one variant is returned per study, if for some reason multiple variants equal to first
+        added=False
         for v in dats[i]:
-            if v.equalize_to(first):
-                res.append(v)
+            if not added:
+                if v.equalize_to(first):
+                    res.append(v)
+                    added=True
+                else:
+                    # Check next variant in case it matches first
+                    if (first.chr == v.chr and first.pos == v.pos):
+                        next_var = s.get_next_data(just_one=True, force_get_next=True)[0]
```

Comment:
I'm not sure what is the problem that this is solving if not the indel situation described?.  All smallest position next variants in the current study is enumerated here as v and compared to the smallest position variant. If the variant v does not match, it is put back into future. If it matches then that will be used in meta.  Now the logic can match multiple variants which is not taken into account originally. So seems that there could have been more variants returned here than actually studies in that case.  To me it seems that all matching ones should be collected first and then if there is a perfect match without swapping, return that and put the other ones back. 

So specifically I don't get how moving to next position solves anything?

Reply:
the original logic should work just fine in multiple variants in the same position except in those insertion and deletion in the same position that just swapping ref/alt make the variants the same.

Reply:
The problem is that `future` does not always include all the variants in the same position. One could be in `future` and other still not read in from the file. In that case the one variant in `future` is returned by `get_next_data` and the next one in the file is not checked.

Example: We have these variants next in the **right study** file:
```
chr  pos  ref  alt
1  100  A  C
1  200  T  G
1  200  C  A
```
Think about a situation where `get_next_data` is called and the `future` is empty. Then, next variant (`1:100:A:C`) is read from the file and added to `vars` and the while-loop starts the next iteration. Now, the next variant (`1:200:T:G`) is read but it's on a different position than the previous and gets thrown in the `future` and the previous variant (`1:100:A:C`) is returned. Now let's say the variant returned is matched in `get_next_variant` and the program proceeds to find next set of variants. Now the situation is that the right study has the one variant on position `1:200` in `future` and the next variant on the same position is waiting to be read. **But**, the logic in `get_next_data` now says that since `future` has elements, it returns something from there (the first variant and the next ones if they match the first) and nothing is read from the actual file at this point. If this returned variant (`1:200:T:G`) is not a correct match (but actually the next one waiting to be read would be the correct one), the situation described in #27 starts and matching is not done correctly.

Now the PR fixes this, but I agree that, like you said, all matching variants should be collected first and the perfect match should be returned and others put back. So far only the next variant in file is checked (`just_one=True`). I'll make adjustments on that.

Reply:
Still don't see it! This calls for call I think!
This is my thinking here.
When a variant is read from a study file, all same position variants will be read and either consumed or put back to the `future`   no exceptions.
In your example key is also what is in the left study. So in here we would be in a scenario in `get_next_variant` that `smallest` is say study_left_1_99_T_G and we read study_right_100_A_C and it's not matching and we put that into the `future`. we would go about processin 1_99_T_G meta and come back to calling `get_next_data`. We are looking then for the next smallest position across studies. Be it newly read from other studies or from anyones future. So what in this scenario is the `first` where we would need to go beyond study_right_1_100_A_C?  study_right_1_100_A_C will be smallest in due time and processed and in the next round study_right_1_200_T_C and study_right_1_200_C_A will be read and will be either directly or via `future` consumed.

Reply:
Let's resolve this in a call!